### PR TITLE
Bugfix/atr 359 vs breaks

### PR DESCRIPTION
**Pull request by SENya1990**
Merge after atr 332 fbql support

### CODE REVIEW

File: docs/dev/CodingGuidelines/CodingGuidelines.md [hunk 1]
```diff
@@ -265,3 +265,20 @@ You should use the `async` / `await` pattern to avoid wrapping exceptions in `Ag
 ```C#
 public async Task TestDiagnostic(string actual) => await VerifyCSharpDiagnosticAsync(actual);
 ```
+
+You must never use the asynchronous delegates in analyzers because they prevent Visual Studio from catching `OperationCancelledException` which leads to VS silent crash without anything being written to the logs.
```
Comment (SENya1990) [inline docs/dev/CodingGuidelines/CodingGuidelines.md]: @k-s-popova please review the new rule in the coding guideline

File: docs/dev/CodingGuidelines/CodingGuidelines.md [hunk 2]
```diff
@@ -265,3 +266,42 @@ You should use the `async` / `await` pattern to avoid wrapping exceptions in `Ag
 ```C#
 public async Task TestDiagnostic(string actual) => await VerifyCSharpDiagnosticAsync(actual);
 ```
+
+### Async anonymous delegates
```
Comment (SENya1990) [inline docs/dev/CodingGuidelines/CodingGuidelines.md]: @k-s-popova please review the modified guide

**Comment by metallmonster** [comment]
@SENya1990 it is very difficult to review this pull-request because, as I can understand, it contains the changes from another pull-request.

**Comment by SENya1990** [comment]
@metallmonster we need to merge previous ones.
@dropsonic  said it's ok to merge without him so I'm going to merge previous unapproved requests

**Comment by SENya1990** [comment]
@metallmonster now it's easier to see and if you approve and merge the previous pull request everything will become even more clear

File: docs/dev/CodingGuidelines/CodingGuidelines.md (hunk 2 above)
Comment (k-s-popova) [reply docs/dev/CodingGuidelines/CodingGuidelines.md]: @SENya1990 , done

**Pull request merged**